exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -18,6 +29,20 @@
   private final String projectId;
   private final Map<String, AttributeValue> fixedAttributes;
 
+  static TraceExporter createWithStub(String projectId, TraceServiceClient traceServiceClient,
+      Map<String, AttributeValue> fixedAttributes) {
+    return new TraceExporter(projectId, traceServiceClient, fixedAttributes);
+  }
+
+  static TraceExporter createWithCredentials(String projectId, Credentials credentials,
+      Map<String, AttributeValue> fixedAttributes, Duration deadline) throws IOException {
+    TraceServiceSettings.Builder builder = TraceServiceSettings.newBuilder()
+        .setCredentialsProvider(FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")));
+    // We only use the batchWriteSpans API in this exporter.
+    builder.batchWriteSpansSettings().setSimpleTimeoutNoRetries(org.threeten.bp.Duration.ofMillis(deadline.toMillis()));
+    return new TraceExporter(projectId, TraceServiceClient.create(builder.build()), fixedAttributes);
+  }
+
   public TraceExporter(String projectId, TraceServiceClient traceServiceClient,
```

Comment:
considered making this ctor private. Didn't do it immediate because I would have to delete a unit test and wanted to get feedback first.

Reply:
nit: Please consider adding a TODO if it doesn't come right after this change.

Reply:
Made it private